internal/security/clusteradapter/cluster_service.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusteradapter
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+
+	"github.com/banzaicloud/pipeline/cluster"
+)
+
+//go:generate mockery -name ClusterGetter -inpkg
+// ClusterGetter restricts the external dependencies for the repository
+type ClusterGetter interface {
```

Comment:
This is never used outside of here, I'd pass the `CommonClusterGetter` directly to the cluster service implementation and use it there. That way, you could get rid of `ClusterGetter`, `MakeClusterGetter`, `clusterGetterAdapter`, `ClusterService` and `Cluster`. `NewClusterService` could return an `AnchoreClusterService` which could be renamed to just `ClusterService`.

Reply:
thx, fixed